internal/backends/winit/glwindow.rs

```diff
@@ -433,6 +433,11 @@ impl<Renderer: WinitCompatibleRenderer + 'static> WindowAdapterSealed for GLWind
             };
 
             let layout_info_h = component.as_ref().layout_info(Orientation::Horizontal);
+            if let Some(window_item) = runtime_window.window_item() {
+                // Setting the width to its preferred size before querying the vertical layout info
+                // is important in case the height depends on the width
+                window_item.width.set(layout_info_h.preferred_bounded());
```

Comment:
Hm, this sets it unconditionally while in the Qt case this only happens when the width is <= 0. Could that make a difference?

Reply:
Qt and winit here diverge quite a bit because Qt use the sizeHint function to compute the default size and other stuff relative to Qt that makes the code quite different.

In the winit case, we will always reset the width and height a few line later (line 470)  so this should be fine to do inconditionaly.

In the Qt case, the apply_window_properties that we call right after does check that the width and height of the item is still 0, before setting to the default size that comes either from sizeHint (preferred size) or from the default WM size or whatever Qt use for default.

So yes, the behavior might be different in some case, but the idea is to somehow respect the default of both the wm or whatever backend we use.  Maybe we could stop doing that and put more code in core that do the initial sizing of the window. but that's hard.

Reply:
I'm thinking of the following scenario:

1. No width and height are set.
2. show() is called, a width and height is set to the preferred size.
3. The user resizes the window by hand.
4. The user clicks on a button that ends up calling `hide()` on the window (say systray)
5. The user clicks on that systray again to show the window again: The size is set to the preferred size with Winit, but not with Qt (because of the <= 0) check.

Not sure it's worth it?